### DISCUSSION TITLE
option to use full path in filedialog?

### DISCUSSION THREAD

**Discussion by rodrigo-j-goncalves** [lxqt/lxqt#2553] [Q&A] [answered]
When I need to open a file in Libreoffice, I see this dialog:
![image](https://github.com/lxqt/lxqt/assets/74656377/fc0683f0-347c-496e-af67-888cdbb641cb)

Note I have this option enabled in the configuration:
![image](https://github.com/lxqt/lxqt/assets/74656377/e94ae5a6-6cc6-46d3-ac3c-ca60cedd2607)

Version: 24.2.2.2 (X86_64) / LibreOffice Community
Build ID: 420(Build:2)
CPU threads: 24; OS: Linux 6.5; UI render: default; VCL: qt5 (cairo+xcb)
Locale: en-US (en_US.UTF-8); UI: en-US
Ubuntu package version: 4:24.2.2 ~ rc2-0ubuntu0.22.04.1 ~ lo1
Calc: CL threaded

---
I guess this is a kind of GTK file dialog / file picker, is that right? 
What I like about that is that one can paste the entire path of the file to be opened/saved and it works. Contrary, in native LXQt filedialog this is not possible (ie, path and file name have to be entered separately in the dialog).

**Is there a way to enable this for LXQt filedialog?** 

I didn't find it in the docs I've read. I am also aware that one can do the opposite: [use LXQt file dialog for GTK applications](https://github.com/lxqt/xdg-desktop-portal-lxqt/blob/master/README.md#use-lxqt-filedialog-in-applications). But as far as I know there is no way to use the full path instead of entering the path and the file name separately. Or maybe there is some other roundabout?

If there is no way to do this currently, would it be possible to request it? IMO it would make no difference for users who routinely chose the path and the file name separately, while it would mean a big difference for users who routinely copy/paste the full path+filename.

**Comment by tsujan** [comment]
> in native LXQt filedialog this is not possible 

It is, in two ways: click inside an empty area of the path-bar, or right click → Edit Path.

**Reply by rodrigo-j-goncalves** [reply]
Maybe I didn't explain correctly. Let's say I want to edit the file `lxqt.conf`.

1. With my filebrowser I go to the location and copy the full path+name: `/home/rjg/.config/lxqt/lxqt.conf`
2. I open featherpad, then press CTRL+O and the file dialog opens
3. What I was hoping for is to paste `/home/rjg/.config/lxqt/lxqt.conf` into the dialog and then press ENTER.

If I follow those 3 steps in featherpad, nothing happens (I tried to copy the full path+name both in the file name or path text boxes).
If I do that in LibreOffice instead, this opens the `lxqt.conf`

Would it be possible to configure filedialog to behave like this? eg. accept a full path+filename in the filename text box?
If not, would it make sense to create a feature request?

**Reply by tsujan** [reply, marked as answer]
No, that's not possible. Regardless of what other file dialogs may do, "File name" is what it says, and it has a popup (called "completer") for finding items quickly. Anything else would be inconsistent, and we don't want to add yet another entry to the dialog.

That being said, I opened https://github.com/lxqt/libfm-qt/issues/962

**Comment by rodrigo-j-goncalves** [comment]
Thank you for considering this.
